scripts/pre_commit_linter.py

```diff
@@ -896,8 +898,129 @@ def _check_html_directive_name(all_files):
     return summary_messages
 
 
+def _check_directive_scope(all_files):
+    """This function checks that all directives have an explicit
+    scope: {} and it should not be scope: true.
+    """
+    print 'Starting directive scope check'
+    print '----------------------------------------'
+    summary_messages = []
+    # Select JS files which need to be checked.
+    files_to_check = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)
+        and filename.endswith('.js')]
+    failed = False
+    summary_messages = []
+    # Use Pyjsparser to parse a JS file as a Python dictionary.
+    parser = pyjsparser.PyJsParser()
+    for filename in files_to_check:
+        with open(filename) as f:
+            content = f.read()
+        # Parse the body of the content as nodes.
+        parsed_nodes = parser.parse(content)['body']
+        for parsed_node in parsed_nodes:
+            # Check the type of the node.
+            if parsed_node['type'] != 'ExpressionStatement':
+                continue
+            # Separate the expression part of the node.
+            expression = parsed_node['expression']
+            # Check whether the expression belongs to a directive.
+            if (expression['type'] != 'CallExpression' or (
```

Comment:
Also pull the conditions out.

Reply:
For this, I had to separate the if clauses because the second condition depends on the first and the third condition depends on the second. 
Thanks!